[release-v2.6] Team 4 - Fix rancher-version and kube-version annotations

PennyScissors opened the pull request.

phillipsj reviewed (changes requested):
Noticing some of these upper limits of 1.22. Is this because this is the old chart? These don't seem correct to me.

PennyScissors commented on packages/rancher-vsphere/rancher-vsphere-cpi/generated-changes/patch/Chart.yaml.patch:
```diff
@@ -0,0 +1,11 @@
+   catalog.cattle.io/os: linux
+   catalog.cattle.io/permits-os: linux,windows
+-  catalog.cattle.io/rancher-version: '>= 2.6.0-0 <= 2.6.100-0'
++  catalog.cattle.io/rancher-version: '>= 2.6.0-0 < 2.7.0-0'
```
Yes

PennyScissors commented:
@phillipsj Are you referring to any specific case? I reviewed these again and they look right to me. The vsphere charts where I updated the kube-version annotation are old and they either have a kubeVersion I matched in the kube-version annotation, or they don't and I added a new one. The windows-gmsa and wins-upgrader is where I need confirmation since I'm not familiar with those charts at all.

PennyScissors commented:
Updated gmsa from < 1.22 to < 1.24 to match Rancher's latest supported k8s version.

PennyScissors commented:
Closed in favor of this PR to consolidate PRs from all teams: https://github.com/rancher/charts/pull/1964
